bundles/auth/org.eclipse.smarthome.auth.oauth2client/src/test/java/org/eclipse/smarthome/auth/oauth2client/internal/cipher/CipherTest.java

```diff
@@ -0,0 +1,135 @@
+package org.eclipse.smarthome.auth.oauth2client.internal.cipher;
+
+import static org.junit.Assert.*;
+
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.security.NoSuchAlgorithmException;
+import java.util.Dictionary;
+import java.util.Hashtable;
+
+import javax.crypto.KeyGenerator;
+import javax.crypto.SecretKey;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.osgi.framework.InvalidSyntaxException;
+import org.osgi.service.cm.Configuration;
+import org.osgi.service.cm.ConfigurationAdmin;
+
+/**
+ * Verify that the SymmetricKeyCipher is working properly.
+ *
+ * This test cannot run until following lines are added to MANIFEST.MF:
+ * junit.framework;resolution:=optional,
+ * org.hamcrest;resolution:=optional,
+ * org.junit;resolution:=optional,
+ *
+ * @author Gary Tse - Initial Contribution
+ *
+ */
+public class CipherTest extends SymmetricKeyCipher {
+
+    private static final String ENCRYPTION_ALGO = "AES/CBC/PKCS5Padding";
+    private static final String PLAIN_TEXT = "hello world";
+
+    @Before
+    public void setUp() {
+        super.setConfigurationAdmin(new ConfigurationAdminMock());
+        super.activate();
+    }
+
+    @Test
+    public void testEncDec() throws GeneralSecurityException {
+
+        String cipherText = encrypt(PLAIN_TEXT);
+        assertNotEquals("Cipher text should not be the same as plaintext", PLAIN_TEXT, cipherText);
+
+        String decryptedText = decrypt(cipherText);
+        assertEquals("Decrypted text should be same as before", PLAIN_TEXT, decryptedText);
+    }
+
+    private static SecretKey generateEncryptionKey() throws NoSuchAlgorithmException {
+        KeyGenerator keygen = KeyGenerator.getInstance("AES");
+        keygen.init(128); // cannot use very high grade encryption due to jvm limit
+        SecretKey secretKey = keygen.generateKey();
+        return secretKey;
+    }
+
+    private static class ConfigurationAdminMock implements ConfigurationAdmin {
+
+        @Override
+        public Configuration createFactoryConfiguration(String factoryPid) throws IOException {
+            return new ConfigurationMock();
+        }
+
+        @Override
+        public Configuration createFactoryConfiguration(String factoryPid, String location) throws IOException {
+            return new ConfigurationMock();
+        }
+
+        @Override
+        public Configuration getConfiguration(String pid, String location) throws IOException {
+            return new ConfigurationMock();
+        }
+
+        @Override
+        public Configuration getConfiguration(String pid) throws IOException {
+            return new ConfigurationMock();
+        }
+
+        @Override
+        public Configuration[] listConfigurations(String filter) throws IOException, InvalidSyntaxException {
+            return new Configuration[] { new ConfigurationMock() };
+        }
+
+    }
+
+    private static class ConfigurationMock implements Configuration {
```

Comment:
Instead of your own custom mock, you can use Mockito which, I believe, already exists in ESH

Reply:
Thx for suggestion.

Reply:
Switched to Mockito, see next commit.